libopae++/include/opaec++/pvalue.h

```diff
@@ -36,6 +38,57 @@ namespace fpga
 namespace types
 {
 
+struct guid_t
+{
+    guid_t(fpga_properties * p)
```

Comment:
Why fpga_properties * ?  fpga_properties is the opaque type which is already a pointer.

Reply:
These property wrapper structs are constructed as members of the `properties` class. They are all initialized with a pointer to the private member `props_` of the `properties` instance which is invalid until the ctor of `properties` which uses `fpgaGetProperties` to allocate the `props_`. That is why all of the property structs need to have a reference (pointer) to `props_`.

Reply:
Ok, that makes sense.